src/main/java/uk/gov/ons/ctp/response/casesvc/message/CaseNotificationPublisher.java

```diff
@@ -1,32 +1,60 @@
 import com.godaddy.logging.Logger;
 import com.godaddy.logging.LoggerFactory;
-import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.integration.annotation.MessageEndpoint;
-import uk.gov.ons.ctp.response.casesvc.message.notification.CaseNotification;
+import org.springframework.stereotype.Service;
+import uk.gov.ons.ctp.response.casesvc.config.AppConfig;
+import uk.gov.ons.ctp.response.casesvc.representation.CaseNotificationDTO;
 
 /**
  * Service implementation responsible for publishing case lifecycle events to notification channel
  */
-@MessageEndpoint
+@Service
 public class CaseNotificationPublisher {
   private static final Logger log = LoggerFactory.getLogger(CaseNotificationPublisher.class);
 
-  @Qualifier("caseNotificationRabbitTemplate")
-  @Autowired
-  private RabbitTemplate rabbitTemplate;
+  @Autowired AppConfig appConfig;
 
+  @Autowired private PubSub pubSub;
+
+  @Autowired private ObjectMapper objectMapper;
   /**
-   * To put one CaseNotification on the outbound channel caseNotificationOutbound
+   * sends case notification to action service via pubsub
    *
-   * @param caseNotification the CaseNotification to put on the outbound channel
+   * @param caseNotificationDTO the CaseNotification to put on the pubsub
    */
-  public void sendNotification(CaseNotification caseNotification) {
-    log.with("case_notification", caseNotification)
-        .debug("Entering sendNotification with CaseNotification");
-    rabbitTemplate.convertAndSend(caseNotification);
+  public void sendNotification(CaseNotificationDTO caseNotificationDTO) {
+    log.with("case_notification", caseNotificationDTO).info("sending CaseNotification to pubsub");
+    try {
+      String message = objectMapper.writeValueAsString(caseNotificationDTO);
+      ByteString data = ByteString.copyFromUtf8(message);
+      PubsubMessage pubsubMessage = PubsubMessage.newBuilder().setData(data).build();
+      Publisher publisher = pubSub.caseNotificationPublisher();
+      try {
+        log.with("publisher", publisher).info("Publishing message to pubsub");
+        ApiFuture<String> messageIdFuture = publisher.publish(pubsubMessage);
+        String messageId = messageIdFuture.get();
+        log.with("messageId", messageId).info("Case Notification sent successfully");
+      } finally {
+        publisher.shutdown();
+        pubSub.shutdown();
+      }
+    } catch (JsonProcessingException e) {
+      log.with("case_notification", caseNotificationDTO)
+          .error("Error while case_notification can not be parsed.");
+      throw new RuntimeException(e);
```

Comment:
Its a parsing error, so I am expecting it case never to reach at this point. But said that. Incase if it happens it will be lost. and there is no point retrying it as it will fail again.

Reply:
True and I probably put this comment in the wrong place, what about the catch statement underneath the interrupted/execution/io exceptions. I guess in those cases we could potentially retry?

Reply:
I thought of that .. but with the existing implementation of rabbit it is the same case. I am not sure how to handle the issue except one thing i.e. publishing message with a retry settings. what do you suggest?